src/staking/phase1/index.ts

```diff
@@ -0,0 +1,349 @@
+    stakingParams: Phase1Params,
+    delegation: Phase1Delegation,
+    feeRate: number,
+  ): PsbtTransactionResult => {
+    const { stakingTx, btcStakingTx } = this.validateAndDecodeDelegaitonInputs(
+      delegation, stakingParams,
+    );
+    // Build scripts
+    const scripts = buildScripts(
+      stakingParams,
+      delegation.finalityProviderPkHex,
+      stakingTx.timelock,
+      delegation.stakerPkHex,
+    );
+
+    // Create the timelock unbonded transaction
+    try {
+      return withdrawTimelockUnbondedTransaction(
+        scripts,
+        btcStakingTx,
+        this.stakerInfo.address,
+        this.network,
+        feeRate,
+        stakingTx.outputIndex,
+      );  
+    } catch (error) {
+      throw StakingError.fromUnknown(
+        error, StakingErrorCode.BUILD_TRANSACTION_FAILURE,
+        "Cannot build unsigned timelock unbonded transaction",
+      );
+    }
+  }
+
+  validateAndDecodeDelegaitonInputs = (
+    delegation: Phase1Delegation, stakingParams: Phase1Params,
+  ) => {
+    const stakingTx = delegation.stakingTx;
+    if (stakingTx.startHeight < stakingParams.activationHeight) {
+      throw new StakingError(
+        StakingErrorCode.INVALID_INPUT,
+        "Staking transaction start height cannot be less than activation height",
+      );
+    } else if (
+      stakingTx.timelock < stakingParams.minStakingTimeBlocks ||
+      stakingTx.timelock > stakingParams.maxStakingTimeBlocks
+    ) {
+      throw new StakingError(
+        StakingErrorCode.INVALID_INPUT,
+        "Staking transaction timelock is out of range",
+      );
+    } else if (delegation.stakerPkHex !== this.stakerInfo.publicKeyHex) {
+      throw new StakingError(
+        StakingErrorCode.INVALID_INPUT,
+        "Staker public key does not match between connected staker and delegation staker",
+      );
+    }
+    let btcStakingTx: Transaction;
+    try {
+      btcStakingTx = Transaction.fromHex(stakingTx.txHex);
+    } catch (error) {
+      throw StakingError.fromUnknown(
+        error, StakingErrorCode.INVALID_INPUT,
+        "Invalid staking transaction hex",
+      );
+    }
+    const btcTxId = Buffer.from(btcStakingTx.getHash()).reverse().toString('hex');
+    if (!btcStakingTx.outs[stakingTx.outputIndex]) {
+      throw new StakingError(
+        StakingErrorCode.INVALID_INPUT,
+        "Staking transaction output index is out of range",
+      );
+    } else if (btcTxId !== delegation.stakingTxHashHex) {
+      throw new StakingError(
+        StakingErrorCode.INVALID_INPUT,
+        "Staking transaction hash does not match between the btc transaction and the provided staking hash",
+      );
+    }
+    return { btcStakingTx, stakingTx };
+  }
+}
+
+const isValidStakerInfo = (stakerInfo: StakerInfo, network: networks.Network): boolean =>
+  isValidBitcoinAddress(stakerInfo.address, network) &&
+  isValidNoCordPublicKey(stakerInfo.publicKeyHex);
+
+const buildScripts = (
```

Comment:
Why did we decide to not include `buildScripts` and `isValidStakerInfo` in the class as we did with the `validateAndDecodeDelegationInputs`?